### PR TITLE
Fixes for Windows-specific compilation issues

### PR DESCRIPTION
**Pull request by livecodefraser**
Fixes the Windows compilation failures encountered when CI-testing https://github.com/livecode/livecode/pull/3017.

### CODE REVIEW

File: util/encode_source.pl [hunk 1]
```diff
@@ -1,11 +1,12 @@
 #!/usr/bin/env perl
 
 use warnings;
+use File::Basename;
 
 # Incoming arguments
 my $sourceFile	= $ARGV[0];
 my $destFile	= $ARGV[1];
-my $varName		= $ARGV[2];
+my $varName	= basename($ARGV[2]);
```
Comment (livecodestephen) [inline util/encode_source.pl:9]: Perhaps an explanation should be included as a comment in the script.  This looks pretty odd otherwise.

Comment (livecodefraser) [inline util/encode_source.pl:9]: Quite right! That's what happens when I decide to turn my quick-fix into a PR ;)

Comment (montegoulding) [inline util/encode_source.pl:9]: I'll add something to save you doing it @livecodefraser ... I've already merged the PR ;-)

Comment (montegoulding) [inline util/encode_source.pl:9]: done

**Pull request merged**